packages/frontend-2/pages/server-management/pending-invitations.vue

```diff
@@ -0,0 +1,272 @@
+        class="rounded-md border border-outline-3"
+      />
+      <div class="flex items-center gap-2 text-foreground text-sm shrink-0">
+        <span class="shrink-0">1-50 of 350</span>
+        <div class="flex gap-1">
+          <ChevronLeftIcon class="h-8 w-8" />
+          <ChevronRightIcon class="h-8 w-8" />
+        </div>
+      </div>
+    </div>
+
+    <Table
+      :headers="[
+        { id: 'email', title: 'Email' },
+        { id: 'invitedBy', title: 'Invited By' },
+        { id: 'resend', title: '' }
+      ]"
+      :items="users"
+      :buttons="[
+        { icon: TrashIcon, label: 'Delete', action: openDeleteInvitationDialog }
+      ]"
+      :column-classes="{
+        email: 'col-span-5',
+        invitedBy: 'col-span-4',
+        resend: 'col-span-3'
+      }"
+    >
+      <template #email="{ item }">
+        {{ item.email }}
+      </template>
+
+      <template #invitedBy="{ item }">
+        <div class="flex items-center gap-2">
+          <img
+            :src="item.profilePicture"
+            :alt="'Profile picture of ' + item.invitedBy"
+            class="w-6 h-6 rounded-full"
+          />
+          {{ item.invitedBy }}
+        </div>
+      </template>
+
+      <template #resend="{ item }">
+        <button class="font-semibold text-primary" @click="resendInvitation(item)">
+          Resend Invitation
+        </button>
+      </template>
+    </Table>
+
+    <DeleteInvitationDialog
+      v-model:open="showDeleteInvitationDialog"
+      :user="user ?? userToModify"
+      title="Delete Invitation"
+      :buttons="[
+        {
+          text: 'Delete',
+          props: { color: 'danger', fullWidth: true },
+          onClick: deleteConfirmed
+        },
+        {
+          text: 'Cancel',
+          props: { color: 'secondary', fullWidth: true, outline: true },
+          onClick: closeDeleteInvitationDialog
+        }
+      ]"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import {
+  MagnifyingGlassIcon,
+  ChevronLeftIcon,
+  ChevronRightIcon
+} from '@heroicons/vue/20/solid'
+
+import Table from '../../components/server-management/Table.vue'
+import DeleteInvitationDialog from '../../components/server-management/DeleteInvitationDialog.vue'
+
+import { TrashIcon } from '@heroicons/vue/24/outline'
+
+definePageMeta({
+  middleware: ['admin']
+})
+
+const userToModify = ref<User | null>(null)
+
+const openDeleteInvitationDialog = (user: User) => {
+  userToModify.value = user
+  showDeleteInvitationDialog.value = true
+}
+
+const closeDeleteInvitationDialog = () => {
+  userToModify.value = null
+  showDeleteInvitationDialog.value = false
+}
+
+const deleteConfirmed = () => {
+  // Implement actual delete logic here
+  console.log('Deleting user:', userToModify.value)
+  showDeleteInvitationDialog.value = false
+  userToModify.value = null
+}
+
+const resendInvitation = (item) => {
+  console.log(item)
+}
+
+const showDeleteInvitationDialog = ref(false)
+
+export interface User {
```

Comment:
1. Don't export things from Vue components
2. Use a type derived from the GQL query result type